packages/components/input-number/src/input-number.vue

```diff
@@ -193,6 +193,13 @@ export default defineComponent({
     }
     const setCurrentValue = (newVal: number | string) => {
       const oldVal = data.currentValue
+      if (props.stepStrictly) {
+        const stepPrecision = getPrecision(props.step)
+        const precisionFactor = 10 ** stepPrecision
+        newVal =
+          (Math.round(newVal / props.step) * precisionFactor * props.step) /
+          precisionFactor
+      }
```

Comment:
@chenxch You did an excellent job, but the code seems duplicated with: [_decrease](https://github.com/element-plus/element-plus/blob/adf1ecf3ebffaf92dc59119fcb0015d8f1607ffe/packages/components/input-number/src/input-number.vue#L173), would you mind encapsulating them?

Reply:
@chenxch Nice! This is a good start for duplication reduction. Thank you for your cooperation! :bow:

Reply:
We can address this in an additional PR, I'll merge this first to fix the bug.